samples/RazorSlices.Samples.WebApp/Program.cs

```diff
@@ -85,6 +86,37 @@
         : TypedResults.NotFound();
 });
 
+
+var htmxTodos = app.MapGroup("/htmx-todo");
+var renderTodoTable = () => Results.Extensions.RazorSlice<Slices.HtmxTodos._Todos, List<Models.Todo>>(Models.Todos.AllTodos);
+htmxTodos.MapGet("/", () => Results.Extensions.RazorSlice<Slices.HtmxTodos.TodoIndex, List<Models.Todo>>(Models.Todos.AllTodos));
+htmxTodos.MapPost("/", ([FromForm] string title) => {
+    var maxId = Models.Todos.AllTodos.Select(x => x.Id).DefaultIfEmpty(0).Max();
+    Models.Todos.AllTodos.Add(new Models.Todo { Id = maxId + 1, Title = title });
+    return renderTodoTable();
+}).DisableAntiforgery();
+htmxTodos.MapPut("/{id:int}/toggle-complete", (int id) => {
+    var todo = Models.Todos.AllTodos.FirstOrDefault(t => t.Id == id);
+    if (todo is null)
+    {
+        return Results.NotFound();
+    }
+
+    todo.IsComplete = !todo.IsComplete;
+    return renderTodoTable();
+});
+htmxTodos.MapDelete("/{id:int}", (int id) =>
+{
+    var todo = Models.Todos.AllTodos.FirstOrDefault(t => t.Id == id);
+    if (todo is null)
+    {
+        return Results.NotFound();
+    }
+
+    Models.Todos.AllTodos.Remove(todo);
+    return renderTodoTable();
+});
```

Comment:
Would be worth moving all the endpoints added for the HTMX sample to a new static class and just calling the method from here. It's getting a bit crowded in here now :smile:

Reply:
Also update the `RazorSlices.Samples.WebApp.Tests` project to hit at least the GET endpoints added so there's some coverage. Bonus points for adding a new test in that project that does an end-to-end test of all these endpoints, e.g. get todos, add a todo, delete a todo, etc.